### PR TITLE
Move Linked Wiki Files to Main Docs Directories

### PR DESCRIPTION
**Pull request by cailafinn**
**Description of work.**
Moves a number of documentation files off of the wiki and into our internal `.rst` structure. This also helps to remove the number of `www.` links in the project, which we are aiming to shrink. 

This could be split into multiple PRs if specialists for the specific parts that these docs cover want to review them separately. This got a little bigger than I expected. 

**To test:**
1. Delete doc-trees
2. Build `docs-html` and/or `docs-qthelp`
3. Check each of the new pages and check all the links work
4. Check that changed links to these new docs go to the right files and work within he QtHelp window in workbench without opening web pages (the link in the dev-docs is an exception to this, but that won't work anyway until merged). 

Partial work on #31340

*This does not require release notes* because **is is only moving docs to our git trackable system. **

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/ISISSANSReductionBackend.rst [hunk 1]
```diff
@@ -756,14 +756,14 @@ interface method (*parse_line* which takes a single line to parse)
 needs to be provided.
 
 For an overview of the user file commands, please see the
-`user file documentation <https://www.mantidproject.org/SANS_User_File_Commands>`_ .
+:ref:`user file documentation <ISIS_SANS_User_File_QRes-ref>`.
```
Comment (SilkeSchomann) [inline dev-docs/source/ISISSANSReductionBackend.rst:759]: This link is missing!

File: dev-docs/source/ISISSANSReductionBackend.rst [hunk 2]
```diff
@@ -756,14 +756,14 @@ interface method (*parse_line* which takes a single line to parse)
...
 
 Director for *ISISCommandInterface* (CLI)
 *****************************************
 
 The *ISISCommandInterface* is used by some of the power users among the instrument scientists. It is
 an efficient way to customize reductions which require small tweaks between different reductions.
-Please consult the `scripting documentation <https://www.mantidproject.org/Scripting_SANS_Reductions>`_
+Please consult the :ref:`scripting documentation <ScriptingSANSReductions>`
```
Comment (SilkeSchomann) [inline dev-docs/source/ISISSANSReductionBackend.rst:766]: I cannot see any link here.

File: docs/source/interfaces/isis_sans/User File Format.rst [hunk 3]
```diff
@@ -660,6 +660,8 @@ Parameters
 **calib_file.nxs**
     The tube calibration file to use.
 
+.. _ISIS_SANS_User_File_QRes-ref:
```
Comment (SilkeSchomann) [inline docs/source/interfaces/isis_sans/User File Format.rst:663]: There does not seem to be a link here.

**Review by SilkeSchomann — COMMENTED** [review]
Most of the links just worked fine but a few seemed to be missing or invisible. 

Also, I've commented on a few typos/grammatical errors. All of them have not been introduced as part of this PR but it might be easy to just correct them.

File: dev-docs/source/ISISSANSReductionBackend.rst (hunk 1 above)
Comment (ConorMFinn) [reply dev-docs/source/ISISSANSReductionBackend.rst:759]: Ah, yes, this one is one that get's handled by the website after merging (I believe, at least from what @DanielMurphy22 and I can work out).

File: dev-docs/source/ISISSANSReductionBackend.rst (hunk 2 above)
Comment (ConorMFinn) [reply dev-docs/source/ISISSANSReductionBackend.rst:766]: I think this is the same as the one above. Defo worth checking with someone who's certain though. I'll ask around

Comment (SilkeSchomann) [reply dev-docs/source/ISISSANSReductionBackend.rst:766]: I agree, it will be like the one above.

**Review by SilkeSchomann — APPROVED** [review]
The links in docs-html and docs-qthelp work as expected. The dev-doc links have to be checked once the PR is merged. I am happy to approve now though.

**Pull request merged**